src/g_text.c

```diff
@@ -1008,11 +1016,16 @@ static void text_getrect(t_gobj *z, t_glist *glist,
 
     if (x->te_type == T_ATOM && x->te_width > 0)
     {
-        int font = glist_getfont(glist);
-        int fontwidth = glist_fontwidth(glist),
-            fontheight = glist_fontheight(glist);
-        width = (x->te_width > 0 ? x->te_width : 6) * fontwidth + 2;
-        height = fontheight + 1; /* borrowed from TMARGIN, etc, in g_rtext.c */
+        width = (x->te_width > 0 ? x->te_width : 6) * glist_fontwidth(glist);
+        height = glist_fontheight(glist);
+        if (glist_getzoom(glist) > 1)
+        {
+            /* zoom margins */
+            width += ATOM_RMARGIN * glist_getzoom(glist);
+            height += ATOM_BMARGIN * glist_getzoom(glist);
+        }
+        else
+            width += ATOM_RMARGIN; height += ATOM_BMARGIN;
```

Comment:
is this intentional? it seems that `height` is incremented regardless of the result of `glist_getzoom()` (but `width` is only incremented if not zoomed)

Reply:
? It's incremented in the line above. The perhaps rather dumb construction here is to avoid the zoom multiplication unless we're actually zooming.

Reply:
the problem is that height gets doubly incremented.
if you properly indent the code it looks like:

~~~
if (glist_getzoom(glist) > 1)
{
    /* zoom margins */
    width += ATOM_RMARGIN * glist_getzoom(glist);
    height += ATOM_BMARGIN * glist_getzoom(glist);
}
else
    width += ATOM_RMARGIN;
height += ATOM_BMARGIN;
~~~
so if we are zooming, then height gets incremented by
 "(ATOM_BMARGIN + 1) * glist_getzoom(glist)"
which seems not to be wanted.
if it *is* wanted, i'd suggest changing the code to:
~~~
if (glist_getzoom(glist) > 1)
{
    /* zoom margins */
    width += ATOM_RMARGIN * glist_getzoom(glist);
    height += (ATOM_BMARGIN + 1) * glist_getzoom(glist);
}
else
{
    width += ATOM_RMARGIN;
    height += ATOM_BMARGIN;
}
~~~

Reply:
Ah yeah, I see what you mean. Fixed in both g_text.c and g_rtext.c. That was the cause of the vertical sizing being too large when zoomed.